src/sota_tools/deploy.cc

```diff
@@ -11,8 +11,16 @@
 #include "treehub_server.h"
 #include "utilities/utils.h"
 
+bool CheckPoolState(const OSTreeObject::ptr &root_object, const RequestPool &request_pool) {
+  if (request_pool.run_mode() == RunMode::kWalkTree || request_pool.run_mode() == RunMode::kPushTree) {
+    return !request_pool.is_idle() && !request_pool.is_stopped();
+  } else {
```

Comment:
Clarity: having an exhaustive switch/case on all existing running modes would make this part easier to follow as it wouldn't need to know about the exact enum cases.